pkg/pillar/containerd/containerd.go

```diff
@@ -229,22 +178,48 @@ func CtrPrepare(containerPath string, ociFilename string) error {
 	log.Infof("mountPoints %+v execpath %+v cmd %+v workdir %+v env %+v",
 		mountpoints, execpath, cmd, workdir, unProcessedEnv)
 
-	if err != nil {
-		return fmt.Errorf("ctrPrepare: unable to get image: %v config: %v", ctrdImage.Name(), err)
-	}
-	// containerd.NewImageWithPlatform(client, i, platforms.Only(platform))
+	// unpack the rootfs Image if needed
 	unpacked, err := ctrdImage.IsUnpacked(ctrdCtx, defaultSnapshotter)
 	if err != nil {
-		return fmt.Errorf("ctrPrepare: unable to get image metadata: %v config: %v", ctrdImage.Name(), err)
+		return fmt.Errorf("snapshotPrepare: unable to get image metadata: %v config: %v", ctrdImage.Name(), err)
 	}
 	if !unpacked {
 		if err := ctrdImage.Unpack(ctrdCtx, defaultSnapshotter); err != nil {
-			return fmt.Errorf("ctrPrepare: unable to unpack image: %v config: %v", ctrdImage.Name(), err)
+			return fmt.Errorf("snapshotPrepare: unable to unpack image: %v config: %v", ctrdImage.Name(), err)
 		}
 	}
-	if err = ctrCreate(filepath.Base(containerPath), ctrdImage); err != nil {
-		return fmt.Errorf("ctrPrepare: failed to create container %s, error: %v", containerPath, err.Error())
+
+	// use rootfs unpacked image to create a writable snapshot with default snapshotter
+	diffIDs, err := ctrdImage.RootFS(ctrdCtx)
+	if err != nil {
+		log.Errorf("Could not load rootfs of image: %v. %v", ctrdImage.Name(), err)
+		return fmt.Errorf("snapshotPrepare: Could not load rootfs of image: %v. %v", ctrdImage.Name(), err)
+	}
+
+	snapshotter := CtrdClient.SnapshotService(defaultSnapshotter)
+	parent := identity.ChainID(diffIDs).String()
+	snapshot := getSnapshotName(containerID)
+	labels := map[string]string{"containerd.io/gc.root": time.Now().UTC().Format(time.RFC3339)}
+	if mounts, err := snapshotter.Prepare(ctrdCtx, snapshot, parent, snapshots.WithLabels(labels)); err != nil {
```

Comment:
We get the mounts here, but do not pass them back. These are needed - or at least the top mount point - to know where to mount up, i.e. where the volume mountpoint is. The single mountpoint for the rootfs should be returned.

Reply:
We ask for that same information when we need it (see getMountPointsFromSnapshot()) -- potentially stashing it someplace would be useful -- but more than I'm willing to put into this bit of refactoring (potentially we can include it in further PRs).